runtime/decimal.c

```diff
@@ -96,6 +113,9 @@ TaggedPtrPanicCode finish(decQuad *dq, decContext *cx) {
             result.panicCode = 0;
             result.ptr = createDecimal(dq);
         }
+        else if (status & DEC_IEEE_754_Invalid_operation) {
+            result.panicCode = PANIC_INVALID_DECIMAL;
+        }
         else {
```

Comment:
This should only happen with DEC_Overflow, right? The code doesn't capture that.

Reply:
Yes, checked the `DEC_Overflow` and moved `DEC_IEEE_754_Invalid_operation` into the else part.

Reply:
Did you mean this check?
```
    if (decQuadClass(dq) == DEC_CLASS_POS_ZERO) {
        decQuadZero(&d);
    } 
```

Reply:
I meant `if (status & DECIMAL_STATUS_FAIL) {`. Is that check redundant now?

Reply:
I think we need to have it for the readability. In `status` these are not only the flags that raise. Except for error flags, we may get inexact and rounded flags. So I think it is better to have the logic like this

```
if (Dec_error) {
// handle each error
}
else {
// No error flag, return value
}
```